llvm/include/llvm/ExecutionEngine/JITLink/JITLinkMemoryManager.h

```diff
@@ -348,6 +348,16 @@ class SimpleSegmentAlloc {
     return Alloc->finalize();
   }
 
+  /// Free allocated memory if finalize won't be called.
+  Error abandon() {
```

Comment:
It would be great to add the missing abondon() function here, because otherwise I cannot use the allocation for fixups (in case of errors, I need to free the memory). The context is easier to understand on the individual commit https://github.com/llvm/llvm-project/pull/169482/commits/f1203f207722a9d95c5dd73b6724308a5ef81f5d